akka-docs/src/main/paradox/typed/persistence-snapshot.md

```diff
@@ -0,0 +1,64 @@
+# Persistence - snapshotting
+
+## Snapshots
+
+As you model your domain using @ref:[persistent actors](persistence.md), you may notice that some actors may be
+prone to accumulating extremely long event logs and experiencing long recovery times. Sometimes, the right approach
+may be to split out into a set of shorter lived actors. However, when this is not an option, you can use snapshots
+to reduce recovery times drastically.
+
+Persistent actors can save snapshots of internal state every N events or when a given predicated of the state
+is fulfilled.
+
+Scala
+:  @@snip [BasicPersistentActorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshottingEveryN }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshottingEveryN }
+
+
+Scala
+:  @@snip [BasicPersistentActorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshottingPredicate }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshottingPredicate }
+
+When a snapshot is triggered incoming commands are stashed until the snapshot has been saved. This means that
+the state can safely be mutable although the serialization and storage of the state is performed asynchronously,
+since the state instance will not be updated by new events until after the snapshot has been saved.
+
+During recovery, the persistent actor is using the previously saved snapshot to initialize the state and then events
+after the snapshot are replayed using the event handler to recover the persistent actor to its current (i.e. latest)
+state.
+
+If not specified, they default to @scala[`SnapshotSelectionCriteria.Latest`]@java[`SnapshotSelectionCriteria.latest()`]
+which selects the latest (youngest) snapshot. It's possible to override the selection of which snapshot to use for
+recovery like this:
+
+Scala
+:  @@snip [BasicPersistentActorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshotSelection }
+
+TODO include corresponding example in Java
+
+To disable snapshot-based recovery, applications can use @scala[`SnapshotSelectionCriteria.None`]@java[`SnapshotSelectionCriteria.none()`].
+A recovery where no saved snapshot matches the specified `SnapshotSelectionCriteria` will replay all journaled
+events. This can be useful if snapshot serialization format has changed in an incompatible way. It should typically
+not be used when events have been deleted.
```

Comment:
What about `deleteMessages` from `EventSourced` with snapshotting? Should we clarify this last sentence?

Reply:
"events have been deleted" corresponds to `deleteMesages`. API for that hasn't been added in Typed yet, issue https://github.com/akka/akka/issues/24698

Note that in untyped persistence, the events are often called messages, which is probably confusing. In Typed we should consistently call them events.